### ISSUE TITLE
Questions while understanding code and paper

### ISSUE THREAD

**Issue by kiki9014** [mingyuliutw/UNIT#7]
I was trying to implement UNIT to tensorflow code

So I read about paper and pytorch code and I have some questions about code.

1. In the paper, there are 4 convolution layers and single FC layer for domain classification in discriminator of domain adaptation from SVHN to MNIST, but code uses dropout layers among convolution layers. Why dropout layers are used in code?

2. In code, feature_losses that computes l1_loss between two features(fake_feat_ab, fake_feat_aa) are used in discriminator loss, but there is no explanation about that loss in paper. Can you explain why these losses are used?

Maybe I missed point in paper, but I can't find out

**Comment by mingyuliutw** [comment]
Thanks for the questions. The arXiv paper is pretty out-of-date. I am going to put the improved papers with more details and improved design before 10/1. I am sorry that some of the details in the code is not discussed in the paper.

For question 1, the dropout layers are used to avoid overfitting for both classifier training and discriminator training.

For question 2, the whole assumption of the work is the shared latent space assumption where we assume if we have a pair of corresponding images, we assume there is a shared latent code. This is why in the discriminator layer for the domain adaptation experiments, we penalize different latent code in the discriminator side.
